rosco-web/src/main/groovy/com/netflix/spinnaker/rosco/controllers/BakeryController.groovy

```diff
@@ -99,8 +99,19 @@ class BakeryController {
     String jobId = jobExecutor.startJob(jobRequest)
 
     // Give the job jobExecutor some time to kick off the job.
+    // Poll for bake status by job id every 1/2 second for 5 seconds.
     // The goal here is to fail fast. If it takes too much time, no point in waiting here.
-    sleep(1000)
+    def startTime = System.currentTimeMillis()
+
+    while (System.currentTimeMillis() - startTime < 5000) {
```

Comment:
I like the idea of having some polling interval and a max timeout. Let's please expose the timeout as a configuration property so it can be tuned. Can do this similar to how `orphanedJobTimeoutMinutes` is handled `BakePoller`. Might as well also expose the interval.